src/manager.rs

```diff
@@ -44,6 +46,24 @@ lazy_static! {
     };
 }
 
+// Workaround the UNC path on Windows, see https://github.com/rust-lang/rust/issues/42869.
+// Otherwise, `Env::from_env()` will panic with error_no(123).
+fn canonicalize_path<'p, P>(path: P) -> io::Result<PathBuf>
+where P: Into<&'p Path> {
+    let canonical = path.into().canonicalize()?;
+    {
+        // TODO: use OsStr instead of Str in case that there are non-utf8 characters
+        // in the path. Pattern matching for OsStr will be implemented by
+        // https://github.com/rust-lang/rust/issues/49802
+        let mut canonical_path = canonical.to_str().unwrap();
+        if canonical_path.starts_with(r"\\?\") {
+            canonical_path = &canonical_path[r"\\?\".len()..];
+            return Ok(PathBuf::from(canonical_path));
```

Comment:
Hmm, I wonder if it would be more robust to unconditionally strip the UNC prefix from paths via the [Url crate](https://doc.servo.org/url/struct.Url.html), i.e. something like:

```rust
Url::from_file_path(canonical).expect("URL").to_file_path().expect("path")
```

I don't know if this would cover more cases or fail less often than the current implementation, however. What do you think?

Reply:
Sure, will check it out.